pep-lint.py

```diff
@@ -0,0 +1,426 @@
+        yield line_num, "Topic must be for a valid sub-index"
+
+
+def _validate_content_type(line_num, line):
+    """'Content-Type' must be 'text/x-rst'"""
+
+    if line != "text/x-rst":
+        yield line_num, "Content-Type must be 'text/x-rst'"
+
+
+def _validate_pep_references(line_num, line):
+    """`Requires`/`Replaces`/`Superseded-By` must be 'NNN' PEP IDs"""
+
+    references = line.split(", ")
+    for reference in references:
+        yield from _pep_num(line_num, reference, "PEP reference")
+
+
+def _validate_created(line_num, line):
+    """'Created' must be a 'DD-mmm-YYYY' date"""
+
+    yield from _date(line_num, line, "Created")
+
+
+def _validate_python_version(line_num, line):
+    """'Python-Version' must be an ``X.Y[.Z]`` version"""
+
+    versions = line.split(", ")
+    for version in versions:
+        if version.count(".") > 2:
+            yield line_num, f"Python-Version must have two or three segments: {version}"
+            continue
+
+        try:
+            major, minor, micro = version.split(".", 2)
+        except ValueError:
+            major, minor = version.split(".", 1)
+            micro = ""
+
+        if major not in "123":
+            yield line_num, f"Python-Version major part must be 1, 2, or 3: {version}"
+        if not _is_digits(minor) and minor != "x":
+            yield line_num, f"Python-Version minor part must be numeric: {version}"
+
+        if micro == "":
+            return
+        if minor == "x":
+            yield line_num, f"Python-Version micro part must be empty if minor part is 'x': {version}"
+        elif micro[0] == "0":
+            yield line_num, f"Python-Version micro part must not have leading zeros: {version}"
+        elif not _is_digits(micro):
+            yield line_num, f"Python-Version micro part must be numeric: {version}"
+
+
+def _validate_post_history(line_num, body):
+    """'Post-History' must be '`DD-mmm-YYYY <Thread URL>`__, …'"""
+
+    body = body
+    if not body:
+        return
+
+    for post in body.replace("\n", " ").removesuffix(",").split(", "):
+        post = post.strip()
+
+        if not post.startswith("`") and not post.endswith(">`__"):
+            yield from _date(line_num, post, "Post-History")
+        else:
+            post_date, post_url = post[1:-4].split(" <")
+            yield from _date(line_num, post_date, "Post-History")
+            yield from _thread(line_num, post_url, "Post-History")
+
+
+def _validate_resolution(line_num, line):
+    """'Resolution' must be a direct thread/message URL"""
+
+    yield from _thread(line_num, line, "Resolution", allow_message=True)
+
+
+########################
+#  Validation Helpers  #
+########################
+
+def _pep_num(line_num, pep_number, prefix):
+    if pep_number.startswith("0") and pep_number != "0":
+        yield line_num, f"{prefix} must not contain leading zeros: {pep_number}"
+    if not _is_digits(pep_number):
```

Comment:
Can we use the builtin?
```suggestion
    if not pep_number.isdigit():
```

https://docs.python.org/3/library/stdtypes.html?highlight=isdigit#str.isdigit

Reply:
The built-in is Unicode-aware --good in most situations, but here we only want ASCII 0-9, hence the helper function. (Probably overkill, but it matches the pygrep lint implementation)

A

Reply:
We can add `and pep_number.isascii()`.